src/typecheck/core.ml

```diff
@@ -29,17 +35,48 @@ module Ae = struct
       | Type.Builtin Ast.False ->
         `Term (Base.app0 (module Type) env s T._false)
 
-      (* Terms connectors *)
+      (* Boolean operators *)
+      | Type.Builtin Ast.Not ->
+        `Term (Base.term_app1 (module Type) env s T.neg)
+      | Type.Builtin Ast.And ->
+        `Term (Base.term_app_left (module Type) env s mk_and)
+      | Type.Builtin Ast.Or ->
+        `Term (Base.term_app_left (module Type) env s mk_or)
+      | Type.Builtin Ast.Xor ->
+        `Term (Base.term_app_left (module Type) env s T.xor)
+      | Type.Builtin Ast.Imply ->
+        `Term (Base.term_app_right (module Type) env s T.imply)
+      | Type.Builtin Ast.Equiv ->
+        `Term (Base.term_app_right (module Type) env s T.equiv)
+
+      (* If-then-else *)
+      | Type.Builtin Ast.Ite ->
+        `Term (
+          Base.make_op3 (module Type) env s (fun _ (a, b, c) ->
+              let cond = Type.parse_prop env a in
+              let then_ = Type.parse_term env b in
+              let else_ = Type.parse_term env c in
+              T.ite cond then_ else_
+          )
+        )
+
+      (* Equality *)
       | Type.Builtin Ast.Eq ->
         `Term (Base.term_app2 (module Type) env s T.eq)
+      | Type.Builtin Ast.Distinct ->
+        `Term (Base.term_app_list (module Type) env s T.distinct)
+
+      (* Ignore the AC symbol *)
```

Comment:
This comment needs to be updated since we don't ignore the AC symbol anymore.

Reply:
Done.